### PR TITLE
Fixes #620 by using tpl function for most probable templated values

### LINKED ISSUE #620: [Helm] Cannot use values based in globals at ingress definition
Helm users most probably will use sealed-secrets as Helm subchart and define later as templates their SealedSecret resources.

When configuring sealed-secrets controller ingress definition there are certain values that will be based on globals to play with values YAMLs hierarchy for example: https://github.com/bitnami-labs/sealed-secrets/blob/main/helm/sealed-secrets/templates/ingress.yaml#L29

If you use subcharting then you most probably you will combine this with [Helm Globals](https://helm.sh/docs/chart_template_guide/subcharts_and_globals/) to avoid repeating the domain in your values file many many times and get some of the power of values overriding for free.

To allow it ingress must start using Helm tpl function.

### PR DESCRIPTION
**Pull request by gcaracuel**
Using tpl function users are allowed now to compound values set in their YAML, specially useful when subcharting.

This might close #620

### CODE REVIEW

**Comment by gcaracuel** [comment]
Is there somebody working on this project?

File: helm/sealed-secrets/templates/ingress.yaml [hunk 1]
```diff
@@ -12,12 +12,16 @@ metadata:
     {{- end }}
   {{- end }}
 spec:
+  {{- if .Values.ingress.tls }}
+  tls:
+  {{ tpl (toYaml .Values.ingress.tls | indent 4) . }}
+  {{- end }}
```
Comment (juan131) [inline helm/sealed-secrets/templates/ingress.yaml:18]: The `.Values.ingress.tls` isn't a map anymore but a boolean, see:
- https://github.com/bitnami-labs/sealed-secrets/blob/main/helm/sealed-secrets/values.yaml#L164
You can use `.Values.ingress.extraTls` to add TLS configuration additional hostname(s), see:
- https://github.com/bitnami-labs/sealed-secrets/blob/main/helm/sealed-secrets/values.yaml#L192

File: helm/sealed-secrets/templates/ingress.yaml [hunk 2]
```diff
@@ -12,12 +12,16 @@ metadata:
...
   {{- if and .Values.ingress.ingressClassName (eq "true" (include "sealed-secrets.supportsIngressClassname" .)) }}
   ingressClassName: {{ .Values.ingress.ingressClassName | quote }}
   {{- end }}
   rules:
-    {{- if .Values.ingress.hostname }}
-    - host: {{ .Values.ingress.hostname }}
+  {{- range .Values.ingress.hosts }}
+    - host: {{ tpl . $ }}
```
Comment (juan131) [inline helm/sealed-secrets/templates/ingress.yaml:24]: The `.Values.ingress.hosts` parameter is deprecated, use `.Values.ingress.extraHosts` instead to add additional hostname(s):
- https://github.com/bitnami-labs/sealed-secrets/blob/main/helm/sealed-secrets/values.yaml#L174

**Review by juan131 — CHANGES_REQUESTED** [review]
Hi @gcaracuel 

Thanks for your PR! Please note your changes are assuming a values structure that isn't valid anymore in the `2.x.x` version of the chart.

**Comment by juan131** [comment]
I'll proceed this PR since it seems it has no activity, please feel free to reopen it.

**Pull request closed without merging**